### ISSUE TITLE
dynamic_calibration

### ISSUE THREAD

**Issue by LinghanQian** [slinshady89/dynamic_distortion_calibrator#10]
Hi all,
i have a new idea, since our project is "dynamic", does that mean when we do the calibration, the system is close-looped system, we should use a controller(e.g. PID) to minimize the error of input and output, and we  can compare in our paper the performance of different controlling algorithms.
@slinshady89 @mekahmedashraf @ViWi92

**Comment by ghost** [comment]
I don't think so. It's not really a close-loop system in the usual sense. The close-loop system was just a way to show to us, what we're supposed to do. Act -> Generate image on screen; Measure -> capture image with camera. Our "controlling algorithm" would then be the pixelwise map between screen and captured image. But we're not actually using a controller, like PID.

**Comment by LinghanQian** [comment]
Okay,  thanks  @ViWi92 , does that mean that all we get at the end is a equation of the mapping ?
Shall we write another function in order to apply the equation on the camera to let it "see the real world"?

**Comment by ghost** [comment]
What we get from `DynamicDistortionCalibrator::findRawDistortion` is two int arrays of the size of the image. So each position in those arrays stands for one camera pixel. For each pixel, we want to know, where on the screen it had been drawn. This is what the arrays tell us, one gives us the x-coordinate of the screen, the other the y-coordinate.
Then we have a function `DynamicDistortionCalibrator::undistort`, which takes in those two maps and the image as captured by the camera. The function then proceeds to create an array large enough to contain the undistorted image. This is likely to be larger than the original image, there was an example for this in the paper about the graycode calibration. Then this new image gets filled with the pixels from the captured image, but at the positions they originally held minus an offset (i.e. the smallest x-index for the x-coordinates, and the smallest y-index for the y-coordinates).

Originally we had planned to also write a function that fits a mathematical function of a higher dimension to each of those maps. That would've meant we'd only have to store a few function parameters (e.g. a function of third order, i&j being the positions of the pixel in the maping arrays: f(i, j) = a*i³ + b*i²y + c*ij² + d*j³ + e -> five parameters). But we don't have time for that anymore.
